test/test_autograd.py

```diff
@@ -1432,6 +1432,349 @@ def backward(self, dy):
             y = MyOp.apply(x)
         self.assertFalse(y.requires_grad)
 
+    def _is_inference_tensor(self, tensor):
+        try:
+            err_msg = "Inference tensor do not track version counter"
+            with self.assertRaisesRegex(RuntimeError, err_msg):
```

Comment:
We have a `is_inference_tensor` method on `TensorImpl` and it can potentially be exposed to `Tensor` and python frontend.(Doesn't have to be in this PR ofc) Wdyt? @albanD